project/part1/Cargo.toml

```
@@ -0,0 +1,40 @@
+[package]
+edition      = "2024"
+name         = "no_std-training"
+rust-version = "1.88"
+version      = "0.1.0"
+
+[[bin]]
+name = "no_std-training"
+path = "./src/bin/main.rs"
+
+[dependencies]
+embassy-executor = "0.9.0"
+embassy-time     = "0.5.0"
+esp-bootloader-esp-idf = { version = "0.4.0", features = [
+  "esp32c3",
+] }
+esp-hal = { version = "1.0.0", features = [
+  "esp32c3",
+  "unstable",
+] }
+esp-println = { version = "0.16.1", features = ["esp32c3"] }
+esp-rtos = { version = "0.2.0", features = [
+  "embassy",
+  "esp32c3",
+] }
+shtcx = { git = "https://github.com/Fristi/shtcx-rs", branch = "feature/async-support" }
+
+[profile.dev]
+# Rust debug is too slow.
+# For debug builds always builds with some optimization
+opt-level = "s"
+
+[profile.release]
+codegen-units    = 1     # LLVM can perform better optimizations using a single thread
+debug            = 2
+debug-assertions = false
+incremental      = false
+lto              = 'fat'
+opt-level        = 's'
+overflow-checks  = false
```

```suggestion
[profile.release]
codegen-units = 1     # LLVM can perform better optimizations using a single thread
debug         = 2
lto           = 'fat'
opt-level     = 's'
```

Some of these release settings are just the default values for this profile, don't think we need to explicitly set these to their existing values.